### PR TITLE
Arabic to Roman

### PR DESCRIPTION
**Pull request by vumanhcuongit**

### CODE REVIEW

File: arabic_to_roman/arabic_to_roman.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      10 => 'X',
+      9 => 'IX',
+      5 => 'V',
+      4 => 'IV',
+      1 => 'I'
+    }
+
+    result = ''
+    symbols.each do |key, value|
+      while self.number >= key
```
Comment (joneslee85) [inline arabic_to_roman/arabic_to_roman.rb]: You can use `number` instead of `self.number`

File: arabic_to_roman/arabic_to_roman.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.number = self.number - key
+        result += value
+      end
+    end
+    return result
```
Comment (joneslee85) [inline arabic_to_roman/arabic_to_roman.rb]: no need to be explicit here, just `result` is okay

File: arabic_to_roman/arabic_to_roman.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
```
Comment (joneslee85) [inline arabic_to_roman/arabic_to_roman.rb]: overall I like it :+1: great work!

File: arabic_to_roman/arabic_to_roman.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  def initialize(arabic)
+    @number = arabic
+  end
+
+  def to_roman
+    result = ''
+    SYMBOLS.each do |key, value|
+      while number >= key
+        self.number = self.number - key
```
Comment (rauxalach) [inline arabic_to_roman/arabic_to_roman.rb]: Whuy you wanna change number value in here ?

File: roman_to_arabic_numeral/roman_to_arabic.rb [hunk 5]
```diff
@@ -1,6 +1,36 @@
 class RomanNumeral
+  def initialize(x)
+    @number = x
+  end
 
-  # code code code
-  # what? why?
+  def to_arabic
+    symbols = {
```
Comment (rauxalach) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: what about bring this up to class CONST ?

File: roman_to_arabic_numeral/roman_to_arabic.rb [hunk 6]
```diff
@@ -1,6 +1,36 @@
 class RomanNumeral
+  def initialize(x)
```
Comment (rauxalach) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: wtf is x ? remember ruby is a very clear and clean language, dont abuse it guys !

File: roman_to_arabic_numeral/roman_to_arabic.rb [hunk 7]
```diff
@@ -1,6 +1,36 @@
...
+      'IV' => 4,
+      'M' => 1000,
+      'D' => 500,
+      'C' => 100,
+      'L' => 50,
+      'X' => 10,
+      'V' => 5,
+      'I' => 1
+    }
+    return symbols[@number] unless symbols[@number].nil?
```
Comment (rauxalach) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: what about ``` return symbols[@number] if symbols[@number] ``` ?

File: roman_to_arabic_numeral/roman_to_arabic.rb [hunk 8]
```diff
@@ -1,6 +1,36 @@
...
 
-end
\ No newline at end of file
+    result = 0
+    while @number != ''
```
Comment (rauxalach) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: Should you better verify @number is not blank or nil at initialize stage ?

File: roman_to_arabic_numeral/roman_to_arabic.rb [hunk 9]
```diff
@@ -1,6 +1,36 @@
...
+      symbols.each do |roman, arabic|
+        if @number.include? roman
+          @number.slice! roman
+          result += symbols[roman]
+          return result += symbols[@number] unless symbols[@number].nil?
+          break
```
Comment (rauxalach) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: how can the program run into break when it was returned ?

File: roman_to_arabic_numeral/roman_to_arabic.rb [hunk 10]
```diff
@@ -1,6 +1,36 @@
...
 
-end
\ No newline at end of file
+    result = 0
+    while @number != ''
+      symbols.each do |roman, arabic|
+        if @number.include? roman
+          @number.slice! roman
+          result += symbols[roman]
+          return result += symbols[@number] unless symbols[@number].nil?
```
Comment (rauxalach) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: again ``` return result += symbols[@number] if symbols[@number] ``` ?

File: arabic_to_roman/arabic_to_roman.rb (hunk 4 above)
Comment (vumanhcuongit) [inline arabic_to_roman/arabic_to_roman.rb]: @rauxalach you mean I should define another var storing number  value and use it from line 27 to line 30 ?

File: roman_to_arabic_numeral/roman_to_arabic.rb (hunk 6 above)
Comment (vumanhcuongit) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: @rauxalach :+1: got it

File: roman_to_arabic_numeral/roman_to_arabic.rb (hunk 5 above)
Comment (vumanhcuongit) [inline roman_to_arabic_numeral/roman_to_arabic.rb]: @rauxalach this pull request contains some old files about "Roman to Arabic" solution, so would you like to review at https://github.com/ruby-vietnam/hacknite/pull/15 ?
Sorry about that.

**Pull request closed without merging**